pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -298,6 +270,80 @@ func (pl *VolumeZone) EventsToRegister() []framework.ClusterEventWithHint {
 	}
 }
 
+// getPersistentVolumeClaimNameFromPod gets pvc names bound to a pod.
+func (pl *VolumeZone) getPersistentVolumeClaimNameFromPod(pod *v1.Pod) []string {
+	var pvcNames []string
+	for i := range pod.Spec.Volumes {
+		volume := pod.Spec.Volumes[i]
+		if volume.PersistentVolumeClaim == nil {
+			continue
+		}
+		pvcName := volume.PersistentVolumeClaim.ClaimName
+		pvcNames = append(pvcNames, pvcName)
+	}
+	return pvcNames
+}
+
+// isSchedulableAfterNodeChange is invoked whenever a node added or updated.
+// It checks whether the change of Node has made a previously unschedulable pod schedulable.
+// If the node label related to pod's pv topology is changed, it could make the pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
```

Comment:
ditto 🙏 

> As we talked in DM, can you refactor this QHint so that it doesn't fetch any other kind of resources? Then, if you think there's nothing we can do for StorageClass, please just update the comment in EventsToRegister. 🙏
https://github.com/kubernetes/kubernetes/pull/124996/files#r1621867140

Reply:
This PR doesn't depend on any other resources.
Node QHint only depends on nodes and the pod.

Reply:
Even though pvcLister is used within `getPVbyPod`, `getPVbyPod` is not a new function and I don't modify this function in this PR.
Therefore I thought using `getPVbyPod` in this function is fine.

Reply:
Whether `getPVbyPod` is a new function introduced in this PR or not doesn't really matter.

If `isSchedulableAfterNodeChange` calls `getPVbyPod` and `getPVbyPod` calls  `pl.pvcLister.PersistentVolumeClaims().Get()`, then `isSchedulableAfterNodeChange` depends on PVC, doesn't it?

Reply:
You are right.
I thought the point is whether the QHint PR increases the complexity of the code.
This PR doesn't introduce getPVbyPod.
Therefore I thought it is OK

Reply:
> Whether getPVbyPod is a new function introduced in this PR or not doesn't really matter.

Sorry, I've missed this message. I will fix it. Thanks!

Reply:
I don't think we can implement `isSchedulableAfterNodeChange` without getPVbyPod.
Therefore, I will close this PR.

Reply:
Yeah, the point is whether QHint would go thru a complicated implementation, that is, fetching another kind of resource(s), rather than how complex a PR is.